### PR TITLE
Feature/report v2 filter

### PR DESCRIPTION
**Pull request by TinyMurky**
### DEVELOP

- [X] Description: 手寫beta版本的三大報表輸出json, 測試report 報表
  
#### Related Issues

- [X] Issue Number:
    - #2776
    - #2825 
#### Checklist

- [x] Used `@` in import paths.
- [x] Verified naming convention compliance ([Naming Convention Guidelines](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md)).
- [x] Coding style verification: checked
- [x] new Library: 0
- [x] new Class / Component: 0
- [x] new loop: 0
- [x] new recursive: 0
- [x] high risk: 0
- [x] new sql: 0

#### UML Diagrams

- None

#### Additional Notes

- Ensured all links are up-to-date.
- Conducted thorough testing on responsive design changes.
- Documented any notable considerations or edge cases addressed.

### CODE REVIEW

File: src/pages/api/v2/company/[companyId]/report/index.test.ts
Comment (TinyMurky) [inline src/pages/api/v2/company/[companyId]/report/index.test.ts:1]: 三大報表的數字正確性在這邊逕行整合測試

File: src/pages/api/v2/company/[companyId]/report/index.test.ts [hunk 1]
```diff
@@ -39,27 +890,406 @@ beforeEach(() => {
...
 
       const { payload, statusMessage } = await handleGetRequest(req, res);
 
       expect(statusMessage).toBe(STATUS_MESSAGE.SUCCESS_GET);
       expect(payload).toBeDefined();
     });
   });
+
+  describe('report handlers', () => {
+    describe('balance sheet handler', () => {
```
Comment (TinyMurky) [inline src/pages/api/v2/company/[companyId]/report/index.test.ts:953]: Balance sheet 測試從這邊開始

File: src/pages/api/v2/company/[companyId]/report/index.test.ts [hunk 2]
```diff
@@ -39,27 +890,406 @@ beforeEach(() => {
...
+        expect(unappropriatedRetainedEarnings).toBeDefined();
+        expect(unappropriatedRetainedEarnings?.curPeriodAmount).toBe(81000);
+      });
+    });
+
+    describe('income statement handler', () => {
+      // Info: (20241017 - Murky) 2024/01/01 ~ 2024/01/31
+      const mockStartDate = timestampInSeconds(1704038400000); // (new Date(2024, 0, 1, 0, 0, 0).getTime());
+      const mockEndDate = timestampInSeconds(1706716799000); // new Date(2024, 0, 31, 23, 59, 59).getTime());
+      it('should calculate correct answer', async () => {
```
Comment (TinyMurky) [inline src/pages/api/v2/company/[companyId]/report/index.test.ts:1101]: Balance sheet 測試從這邊開始

File: src/pages/api/v2/company/[companyId]/report/index.test.ts [hunk 3]
```diff
@@ -39,27 +890,406 @@ beforeEach(() => {
...
+         * Info: (20241017 - Murky)
+         * @description 8200 本期淨利
+         */
+        const profitLoss = general.find((detail) => detail.code === '8200');
+        expect(profitLoss).toBeDefined();
+        expect(profitLoss?.curPeriodAmount).toBe(77000);
+      });
+    });
+
+    describe('cash flow statement handler', () => {
```
Comment (TinyMurky) [inline src/pages/api/v2/company/[companyId]/report/index.test.ts:1227]: cashflow 測試從這邊開始

**Review by Luphia — APPROVED** [review]
- no bug found
